### ISSUE TITLE
NPCs despawn too quickly.

### ISSUE THREAD

**Issue by zana244** [Project-Epoch/bugtracker#8320]
The issue is mainly relevant in dungeons. Out of the ones I tested (mainly ZF/ST), all the NPC corpses despawn after exactly 5 minutes, even if they still have loot on them, even if they are trash or bosses.
At least for the dungeon NPCs (especially bosses), this is horrible, since if you manage to kill them and you wipe, you will lose your loot during your corpse run...
I tried to time some gnomeregan elites on Stormforge, 1 ooze and the Viscous Fallout boss, and after 24 minutes they still had not despawned, but then a server restart came. Testing at the moment for a longer period of time Electrocutioner 6000 + Peacekeeper Security Suit + Leprous Assistant for boss + elite trash + normal trash timers.

As I was writing this, the normal trash despawned in 5 minutes, the elite + boss are still there.
Later edit: Still counting, The elite NPC despawned after 1 hour and the boss is still there as a corpse.
128 min in the boss is still there, seems like stormforge opted for extremely high despawn timers for elite NPCs in dungeons.
On VMangos deadmines however, the goblin woodcarvers + Sneed despawned after 10 minutes, even with loot on them.

I was curious about outdoors NPCs as well. On Epoch, a normal mob will despawn in about 2.5 min (if I timed right) if it has no loot, or in 5 min if it has loot -- most mobs have 5 min respawn timers anyway. It did mostly the same thing on Everlook (VMangos).

I then wanted to check elites.

However, when I killed Mor'ladim on Everlook and did not loot it, the corpse stayed there for the full 10 minutes, and the respawn came in at the same time the corpse despawned. If I looted Mor'ladim, his corpse would despawn in about 3:45 min.

On Epoch, he despawns exactly after 5 min, regardless of whether he has loot or not.

TLDR: Dungeon Elites and outdoors elites need to take longer to despawn their corpses (if they have loot on them)

**Comment by Shadownflame** [comment]
Post fix behavior is rather interesting. Unlooted boss despawn in dungeons is up to 10min from 5 min (this is still too short imo), same as trash mobs. However, looting the mob extended this timer to 12minutes for a regular trash mob (probably started a new timer when I looted...?)
This is an improvement, but functionally not satisfactory. An unlooted enemy should stay longer than a looted one... and Bosses should be very least considered rare-elites, which run on 20minutes as per the new update.

**Comment by Shadownflame** [comment]
Another test made: killed 3 enemies at the same time (1-1second differences)
elite mob I looted immediately after killing despawned after just 8-9minutes
elite mob I did not loot at all despawned after 10min sharp
elite mob I looted ~ 5 minutes after killing it despawned after 13minutes
Note, these are equal status trash mobs inside ZF dungeon. There is no reason why they should be running on different despawn, nor why looting an enemy delays its despawn but loot being on them doesn't...

**Comment by Shadownflame** [comment]
Resolved.
